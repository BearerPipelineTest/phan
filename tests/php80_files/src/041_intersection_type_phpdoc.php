<?php

/**
 * @param callable&object $a
 * @param callable&array<string> $b
 * @param callable&string $c
 */
function test41($a, $b, $c) {
    '@phan-debug-var $a, $b, $c';
    var_dump($a, $b, $c);
    $a();
    $b();
    $c();
}
class A41 {
    public static function method(): void {
        echo "In method\n";
    }
}
test41(function () { echo "."; }, [A41::class, 'method'], 'A41::method');
// wrong arguments
test41([A41::class, 'method'], 'A41::method', function () { echo "."; });
