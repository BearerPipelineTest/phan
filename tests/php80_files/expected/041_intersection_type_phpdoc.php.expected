%s:8 PhanDebugAnnotation @phan-debug-var requested for variable $a - it has union type callable|object
%s:8 PhanDebugAnnotation @phan-debug-var requested for variable $b - it has union type callable|string[]
%s:8 PhanDebugAnnotation @phan-debug-var requested for variable $c - it has union type callable|string
%s:16 PhanUnreferencedPublicMethod Possibly zero references to public method \A41::method()
